### PR TITLE
ci: introduce ruff in place of flake8 for improved speed

### PR DESCRIPTION
**Pull request by vinitkumar**
Replace flake8 with ruff as it is mostly compatible but offer about 50x speedup. It also aims to replace many other tools like isort, flake8 plugins with one single tool.

Also, fixes some code quality checks suggested by ruff

### CODE REVIEW

File: cms/test_utils/testcases.py [hunk 1]
```diff
@@ -433,23 +434,41 @@ def update(self, response):
...
+
     def failUnlessWarns(self, category, message, f, *args, **kwargs):
         warningsShown = []
-        cleanwarningsShown = []
         result = _collectWarnings(warningsShown.append, f, *args, **kwargs)
 
         if not warningsShown:
             self.fail("No warnings emitted")
-
-        for warning in warningsShown:
```
Comment (marksweb) [inline cms/test_utils/testcases.py:444]: How come this has been removed?

File: cms/utils/compat/forms.py [hunk 2]
```diff
@@ -33,9 +33,9 @@
         else:
             from django.contrib.auth.forms import UserChangeForm
     except ImportError:
-        from django.contrib.auth.forms import UserChangeForm  # nopyflakes
-        from django.contrib.auth.forms import UserCreationForm  # nopyflakes
+        from django.contrib.auth.forms import UserChangeForm  # noqa
```
[the remark this answers is not in the document]
Comment (marksweb) [reply cms/utils/compat/forms.py:36]: If all of these are unused - can we remove them while they've first been flagged?

File: ruff.toml [hunk 3]
```diff
@@ -0,0 +1,20 @@
+exclude = [
```
Comment (marksweb) [inline ruff.toml:1]: Would rather we have a `pyproject.toml` to start bringing configs together in 1 place I think.

Comment (vinitkumar) [reply ruff.toml:1]: Agreed.

File: ruff.toml [hunk 4]
```diff
@@ -5,6 +5,7 @@ exclude = [
 ]
 ignore = [
```
[the remark this answers is not in the document]
Comment (vinitkumar) [reply ruff.toml]: Done!

**Comment by marksweb** [comment]
@vinitkumar I think it's worth adding the `I` job so we can remove isort as well.

File: cms/test_utils/testcases.py [hunk 5]
```diff
@@ -433,23 +434,41 @@ def update(self, response):
         request._messages = MockStorage()
         return request
 
+    # TODO: Renable this.
```
Comment (fsbraun) [inline cms/test_utils/testcases.py]: I believe this is an outdated change. The tests should not be affected.

**Review by fsbraun — CHANGES_REQUESTED** [review]
Probably the changes to the `testcases.py` file are not meant to be part of this PR. They seem outdated. Can you remove them?

Otherwise I feel that readability improves a lot. 👍

File: cms/api.py [hunk 6]
```diff
@@ -437,7 +437,7 @@ def publish_pages(include_unpublished=False, language=None, site=None):
         qs = qs.filter(node__site=site)
 
     output_language = None
-    for i, page in enumerate(qs):
+    for _i, page in enumerate(qs):
```
Comment (fsbraun) [inline cms/api.py:440]:
```suggestion
    for page in qs:
```

File: cms/toolbar/toolbar.py [hunk 7]
```diff
@@ -191,7 +191,7 @@ def init_toolbar(self, request, request_path=None):
             self.clipboard = user_settings.clipboard
 
         if hasattr(self, 'toolbars'):
-            for key, toolbar in self.toolbars.items():
+            for key, _toolbar in self.toolbars.items():
```
Comment (fsbraun) [inline cms/toolbar/toolbar.py:194]:
```suggestion
            for key in self.toolbars:
```

**Review by fsbraun — APPROVED** [review]
I am impressed how many clarifications ruff found! Two suggestions on how to simplify to code even more once ruff identified unused loop variables. LGTM. With respect to the CodeQL warnings I believe we have to acknowledge them in the UI since CodeQL does not respect `# pramga` comments etc.

File: cms/utils/compat/forms.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (vinitkumar) [reply cms/utils/compat/forms.py:38]: Removing them breaks it because they are imported somewhere else. I think we can address them in some other PR.

**Pull request merged**